src/plugins/data_source_management/public/components/utils.test.ts

```diff
@@ -219,4 +226,50 @@ describe('DataSourceManagement: Utils.ts', () => {
     /* True cases: port number scenario*/
     expect(isValidUrl('http://192.168.1.1:1234/')).toBeTruthy();
   });
+
+  describe('Check default auth method', () => {
+    test('default auth method is Username & Password when Username & Password is enabled', () => {
+      const authMethodCombinationsToBeTested = [
+        [usernamePasswordAuthMethod],
+        [sigV4AuthMethod, usernamePasswordAuthMethod],
+        [noAuthCredentialAuthMethod, usernamePasswordAuthMethod],
+        [noAuthCredentialAuthMethod, sigV4AuthMethod, usernamePasswordAuthMethod],
+      ];
+
+      authMethodCombinationsToBeTested.forEach((authOptions) => {
+        const authenticationMethodRegistery = new AuthenticationMethodRegistery();
+
+        authOptions.forEach((authMethod) => {
+          authenticationMethodRegistery.registerAuthenticationMethod(authMethod);
+        });
+
+        expect(getDefaultAuthMethod(authenticationMethodRegistery)?.name).toBe(
+          AuthType.UsernamePasswordType
+        );
+      });
+    });
+
+    test('default auth method is first one in AuthList when Username & Password is not enabled', () => {
+      const authMethodCombinationsToBeTested = [
+        [sigV4AuthMethod],
+        [noAuthCredentialAuthMethod],
+        [sigV4AuthMethod, noAuthCredentialAuthMethod],
+      ];
+
+      authMethodCombinationsToBeTested.forEach((authOptions) => {
+        const authenticationMethodRegistery = new AuthenticationMethodRegistery();
+
+        authOptions.forEach((authMethod) => {
+          authenticationMethodRegistery.registerAuthenticationMethod(authMethod);
+        });
+
+        expect(getDefaultAuthMethod(authenticationMethodRegistery)?.name).toBe(authOptions[0].name);
+      });
+    });
+
+    test('default auth method is NoAuth when no auth options configured', () => {
```

Comment:
I thought that will nothing return, and gray out. so we changed the plan to show NoAuth

Reply:
Yup, discussed with @bandinib-amzn, when nothing returned we will show NoAuth and grey out the selector.

Reply:
But this scenario rarely happens because, from @bandinib-amzn, username and password will always been enabled.

Reply:
@bandinib-amzn and I have a further discussion, when there are no option enabled, the UX will looks like this:
![Screenshot 2024-02-26 at 5 11 36 PM](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/139305463/70308ecb-6c1e-4e1d-956e-90eedff9d302)